### PR TITLE
[FEATURE] customizable output width and height in save as image dialog

### PR DESCRIPTION
**Pull request by nirvn**
## Description
This adds output width and height spin boxes to the recently improved save as image feature:
screenshot from 2017-05-01 11-40-26

Wallpaper time!

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nirvn** [comment]
@nyalldawson , world file creation (re-)added.

File: src/core/qgsmaprenderertask.cpp [hunk 1]
```diff
@@ -154,6 +156,38 @@ bool QgsMapRendererTask::run()
       mError = ImageSaveFail;
       return false;
     }
+
+    if ( mSaveWorldFile )
+    {
+      QString content;
+      // note: use 17 places of precision for all numbers output
```
Comment (nyalldawson) [inline src/core/qgsmaprenderertask.cpp]: I don't like this code being here - I think I should be pulled out to a function which can be reused elsewhere. I'd suggest a static QgsMapSettingsUtils::worldFileContent( const QgsMapSettings &settings). What do you think?

Comment (nirvn) [inline src/core/qgsmaprenderertask.cpp]: @nyalldawson , it's a brilliant idea -- I didn't like having this both in qgsmapcanvas.cpp and there.

**Comment by nirvn** [comment]
@nyalldawson , QgsMapSettingsUtils::worldFileContent( ... ) added.

**Comment by nyalldawson** [comment]
Looks good! Do you mind adding some tests for worldFileContent?

**Comment by nirvn** [comment]
@nyalldawson , a test has been added for worldFileContent.

**Pull request merged**